utils/data_utils.py

```diff
@@ -16,130 +16,146 @@
-        # Enter the container and return first element
-        # if it represents an array
-        if len(list(project_client)) == 1:
-            sequence_client = project_client.values()[0]
-            if isinstance(sequence_client, ArrayClient):
-                return sequence_client
-    return None
-
-
-def get_data_shape_by_name(project_name):
-    """
-    Retrieve shape of the data
-    """
-    project_container = get_data_sequence_by_name(project_name)
-    if project_container:
-        return project_container.shape
-    return None
-
-
-def get_annotated_segmented_results(json_file_path="exported_annotation_data.json"):
-    annotated_slices = []
-    with open(json_file_path, "r") as f:
-        for line in f:
-            if line.strip():
-                json_data = json.loads(line)
-                json_data["data"] = json.loads(json_data["data"])
-                annotated_slices = list(json_data["data"][0]["annotations"].keys())
-    return annotated_slices
-
-
-def DEV_load_exported_json_data(file_path, USER_NAME, PROJECT_NAME):
-    """
-    This function is used to load the exported json, which was saved in a local file.
-    User name is used to filter the data by user.
-    Project name is used to filter the data by project.
-    It is sorted by timestamp with the latest timestamp first.
-
-    TODO: this function will be replaced with a database query in the future.
-    """
-    DATA_JSON = []
-
-    with open(file_path, "r") as f:
-        for line in f:
-            if line.strip():
-                json_data = json.loads(line)
-                json_data["data"] = json.loads(json_data["data"])
-                DATA_JSON.append(json_data)
-
-    data = [
-        data
-        for data in DATA_JSON
-        if data["user"] == USER_NAME and data["source"] == PROJECT_NAME
-    ]
-    if data:
-        data = sorted(data, key=lambda x: x["time"], reverse=True)
-
-    return data
-
-
-def DEV_filter_json_data_by_timestamp(data, timestamp):
-    return [data for data in data if data["time"] == timestamp]
-
-
-def save_annotations_data(global_store, all_annotations, project_name):
-    """
-    Transforms annotations data to a pixelated mask and outputs to
-    the Tiled server
-
-    # TODO: Save data to Tiled server after transformation
-    """
-    annotations = Annotations(all_annotations, global_store)
-    annotations.create_annotation_mask(sparse=True)  # TODO: Check sparse status
-
-    # Get metadata and annotation data
-    metadata = annotations.get_annotations()
-    mask = annotations.get_annotation_mask()
-
-    # Get raw images associated with each annotated slice
-    img_idx = list(metadata.keys())
-    img = data[project_name]
-    raw = []
-    for idx in img_idx:
-        ar = img[int(idx)]
-        raw.append(ar)
-    try:
-        raw = np.stack(raw)
-        mask = np.stack(mask)
-    except ValueError:
-        return "No annotations to process."
-
-    return
+
```

Comment:
This is a little picky, but cab we call this class something that links it to tiled, like maybe `TiledDataLoader`

Reply:
I like that!

Reply:
I like this idea too. Now that I am thinking about it, I think it aligns with what Tanny or Wiebke was thinking about implementing a data loader in the future.